deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/BaseProcessor.kt

```diff
@@ -0,0 +1,66 @@
+package com.airbnb.deeplinkdispatch
+
+import androidx.room.compiler.processing.XProcessingEnv
+import androidx.room.compiler.processing.XRoundEnv
+import androidx.room.compiler.processing.XTypeElement
+import androidx.room.compiler.processing.compat.XConverters.toXProcessing
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.processing.SymbolProcessor
+import com.google.devtools.ksp.processing.SymbolProcessorEnvironment
+import com.google.devtools.ksp.symbol.KSAnnotated
+import javax.annotation.processing.AbstractProcessor
+import javax.annotation.processing.ProcessingEnvironment
+import javax.annotation.processing.RoundEnvironment
+import javax.lang.model.SourceVersion
+import javax.lang.model.element.TypeElement
+
+/**
+ * Creates a unified abstraction for processors of both KSP and java annotation processing.
+ */
+abstract class BaseProcessor(val symbolProcessorEnvironment: SymbolProcessorEnvironment?) :
+    AbstractProcessor(),
+    SymbolProcessor {
+
+    lateinit var environment: XProcessingEnv
+        private set
+
+    override fun getSupportedSourceVersion(): SourceVersion = SourceVersion.RELEASE_8
+
+    final override fun init(processingEnv: ProcessingEnvironment) {
+        super<AbstractProcessor>.init(processingEnv)
+        environment = XProcessingEnv.create(processingEnv)
+    }
+
+    final override fun process(annotations: Set<TypeElement>, roundEnv: RoundEnvironment): Boolean {
+        if (roundEnv.errorRaised()) {
```

Comment:
I guess this was from the erf processor -- I can't say I really understand why we want to make these calls. It seems to imply that KAPT needs to communicate errors/finish calls to KSP. Maybe this explains why we couldnt' run both kapt and KSP. @elihart Do you remember why this was necessary?

Reply:
@BenSchwab KSP's `SymbolProcessor` interface contains `onError` and I set up the base processor so that KAPT calls that function upon error as well so that someone subclassing this base processor can override a single function and have it work for both KSP and KAPT to avoid having behavior differ between the two backends